modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppRestSdkClientCodegen.java

```diff
@@ -176,6 +205,19 @@ public void processOpts() {
             defaultInclude = additionalProperties.get(DEFAULT_INCLUDE).toString();
         }
 
+        if (additionalProperties.containsKey(SUPPORTING_FILES_DIRECTORY)) {
+            supportingFilesDirectory = additionalProperties.get(SUPPORTING_FILES_DIRECTORY).toString();
+
+        }
+        if (additionalProperties.containsKey(SUPPORTING_FILES_NAMESPACE)) {
+            supportingFilesNamespace = additionalProperties.get(SUPPORTING_FILES_NAMESPACE).toString();
+        }
+
+        // Ensure that the supportingFilesNamespace always ends with "::"
+        if(!supportingFilesNamespace.isEmpty() && !supportingFilesNamespace.endsWith("::")) {
```

Comment:
I'm wondering why not add the :: in the templates ? The `{{{supportingFilesNamespace}}}ApiClient` was unexpected, I think.